packages/service-clients/odsp-client/src/odspContainerServices.ts

```diff
@@ -3,25 +3,65 @@
  * Licensed under the MIT License.
  */
 
+import { TypedEventEmitter } from "@fluid-internal/client-utils";
 import type { IContainer } from "@fluidframework/container-definitions/internal";
+import type { IDisposable } from "@fluidframework/core-interfaces";
 import { createServiceAudience } from "@fluidframework/fluid-static/internal";
 
 import type {
 	IOdspAudience,
 	OdspContainerServices as IOdspContainerServices,
+	IOdspContainerServicesEvents,
 } from "./interfaces.js";
 import { createOdspAudienceMember } from "./odspAudience.js";
 
 /**
  * @internal
  */
-export class OdspContainerServices implements IOdspContainerServices {
+export class OdspContainerServices
+	extends TypedEventEmitter<IOdspContainerServicesEvents>
+	implements IOdspContainerServices, IDisposable
+{
+	private _disposed = false;
```

Comment:
Nit: we can probably leverage an ES22 private property here, rather than using `_` naming.
```
	#disposed = false;
```

Reply:
Same for the other `private` members